Services/User/Profile/classes/class.ilPersonalProfileGUI.php

```diff
@@ -527,7 +527,61 @@ protected function showUserAgreement()
         $this->tpl->setPermanentLink('usr', null, 'agreement');
         $this->tpl->printToStdout();
     }
-    
+
+    protected function showConsentWithdrawalConfirmation() : void
+    {
+        $this->tabs->clearTargets();
+        $this->tabs->clearSubTabs();
+        $this->tpl->setTitle($this->lng->txt('withdraw_consent'));
+
+        $tpl = new \ilTemplate('tpl.withdraw_terms_of_service.html', true, true, 'Services/TermsOfService');
+        $tpl->setVariable('TERMS_OF_SERVICE_WITHDRAWAL_CONTENT', $this->lng->txt('withdraw_consent_info'));
+        $tpl->setVariable('FORM_ACTION', $this->ctrl->getFormAction($this));
+
+        $tpl->setVariable('WITHDRAW_TERMS_OF_SERVICE', $this->lng->txt('withdraw_usr_agreement'));
+        $tpl->setVariable('TXT_WITHDRAW', $this->lng->txt('withdraw'));
+        $tpl->setVariable('TXT_CANCEL', $this->lng->txt('cancel'));
+
+        $this->tpl->setContent($tpl->get());
+        $this->tpl->setPermanentLink('usr', null, 'agreement');
+        $this->tpl->printToStdout();
+    }
+
+    protected function cancelWithdrawal() : void
+    {
+        $this->user->writePref('consent_withdrawal_requested', 0);
+        ilInitialisation::redirectToStartingPage();
+    }
+
+    protected function withdrawAcceptance() : void
+    {
+        global $DIC;
```

Comment:
Please don't use the `global` keyword here (it is not a static method, so this should be doable). Instead, use the `$DIC` only in `__construct`, fetch the required services from the container, and assign them to member variables.